makefile

```diff
@@ -5,7 +5,27 @@ IMPORTS =
 ROKU_TEST_ID = 1
 ROKU_TEST_WAIT_DURATION = 5
 
+MAJOR = $(shell grep major_version manifest | sed 's/major_version=//')
+MINOR = $(shell grep minor_version manifest | sed 's/minor_version=//')
+BUILD_NUM = $(shell grep build_version manifest | sed 's/build_version=//' | bc)
+BUILD_HASH = $(shell git log -n1 --format=format:"%h")
+# BRANCH_NAME is set by Jenkins. When on a release branch, the following will evaluate to `rc`, while on a develop branch, it will be left alone (`develop`).
+RC_DEVELOP = $(shell echo $$BRANCH_NAME | sed "s/release.*/rc/")
+
 ZIP_EXCLUDE = -x *.sh -x makefile -x dist\* -x *app.mk* -x *README* -x *rokuTarget* -x *.svn* -x *.git* -x *.DS_Store* -x out\* -x packages\* -x design\* -x node_modules/**\* -x node_modules -x .buildpath* -x .project* -x renderer\* -x backup\* -x *.code-workspace
 
 APPSROOT = .
 include $(APPSROOT)/app.mk
+
+# deploy `TruexRokuReferenceApp` side-load capable zip file to s3
+# append the major, minor then rc or develop components to the upload path.
+# Note: the MAJOR, MINOR, BUILD_NUM, BUILD_HASH env variables are set upstream by the Jenkins system (TAR's Jenkinsfile)
+deploy: $(APPNAME)
+	S3_BUCKET=$$(grep s3_bucket ~/Library/Truex/Roku/target | sed 's/s3_bucket=//') ;\
+	echo $$S3_BUCKET ;\
+	echo $$MAJOR ;\
+	echo $$MINOR ;\
+	echo $$BUILD_NUM ;\
+	echo $$BUILD_HASH ;\
+	aws s3 cp dist/apps/${APPNAME}.zip s3://$$S3_BUCKET/roku/v${MAJOR}_${MINOR}/${RC_DEVELOP}/${APPNAME}-${RC_DEVELOP}-v${MAJOR}.${MINOR}.${BUILD_NUM}-${BUILD_HASH}.zip --acl public-read ;\
```

Comment:
Remember that for the reference app we want to update the component library reference in `MainScene.xml` to point to the signed TAR build that is produced earlier on in the pipeline.

I think it should be straightforward to add this URL substitution logic right in the makefile, using sed commands or similar. We have done something along these lines for the `TruexConstants.brs` file update.